test/workbox-google-analytics/node/test-index.mjs

```diff
@@ -0,0 +1,338 @@
+    reset();
+    sandbox.stub(_private.logger);
+  });
+
+  after(function() {
+    process.removeListener('unhandledRejection', catchUnhandledPromiseRejections);
+    reset();
+  });
+
+  it(`should register a handler to cache the analytics.js script`, function() {
+    sandbox.spy(NetworkFirst.prototype, 'handle');
+
+    googleAnalytics.initialize();
+
+    self.dispatchEvent(new FetchEvent('fetch', {
+      request: new Request(
+          `https://${GOOGLE_ANALYTICS_HOST}${ANALYTICS_JS_PATH}`, {
+        mode: 'no-cors',
+      }),
+    }));
+
+    expect(NetworkFirst.prototype.handle.calledOnce).to.be.true;
+  });
+
+  it(`should register GET/POST routes for /collect`, function() {
+    sandbox.spy(NetworkOnly.prototype, 'handle');
+
+    googleAnalytics.initialize();
+
+    self.dispatchEvent(new FetchEvent('fetch', {
+      request: new Request(`https://${GOOGLE_ANALYTICS_HOST}` +
+          `${COLLECT_PATH}?${PAYLOAD}`, {
+        method: 'GET',
+      }),
+    }));
+
+    expect(NetworkOnly.prototype.handle.calledOnce).to.be.true;
+
+    self.dispatchEvent(new FetchEvent('fetch', {
+      request: new Request(`https://${GOOGLE_ANALYTICS_HOST}` +
+          `${COLLECT_PATH}`, {
+        method: 'POST',
+        body: PAYLOAD,
+      }),
+    }));
+
+    expect(NetworkOnly.prototype.handle.calledTwice).to.be.true;
+  });
+
+  it(`should not alter successful hits`, async function() {
+    sandbox.stub(self, 'fetch');
+
+    googleAnalytics.initialize();
+
+    self.dispatchEvent(new FetchEvent('fetch', {
+      request: new Request(`https://${GOOGLE_ANALYTICS_HOST}` +
+          `${COLLECT_PATH}?${PAYLOAD}`, {
+        method: 'GET',
+      }),
+    }));
+
+    expect(self.fetch.calledOnce).to.be.true;
+    expect(self.fetch.firstCall.args[0].url).to.equal(`https://` +
+        `${GOOGLE_ANALYTICS_HOST}${COLLECT_PATH}?${PAYLOAD}`);
+
+    self.dispatchEvent(new FetchEvent('fetch', {
+      request: new Request(`https://${GOOGLE_ANALYTICS_HOST}` +
+          `${COLLECT_PATH}`, {
+        method: 'POST',
+        body: PAYLOAD,
+      }),
+    }));
+
+    expect(self.fetch.calledTwice).to.be.true;
+
+    const bodyText = await self.fetch.secondCall.args[0].text();
+    expect(bodyText).to.equal(PAYLOAD);
+  });
+
+  it(`should add failed hits to a background sync queue`, async function() {
+    sandbox.stub(self, 'fetch').rejects(new TestError());
+    sandbox.spy(Queue.prototype, 'addRequest');
+
+    googleAnalytics.initialize();
+
+    self.dispatchEvent(new FetchEvent('fetch', {
+      request: new Request(`https://${GOOGLE_ANALYTICS_HOST}` +
+          `${COLLECT_PATH}?${PAYLOAD}`, {
+        method: 'GET',
+      }),
+    }));
+
+    self.dispatchEvent(new FetchEvent('fetch', {
+      request: new Request(`https://${GOOGLE_ANALYTICS_HOST}` +
+          `${COLLECT_PATH}`, {
+        method: 'POST',
+        body: PAYLOAD,
+      }),
+    }));
+
+    await waitUntil(() => Queue.prototype.addRequest.callCount === 2);
```

Comment:
I think there might be a creative solution to this that doesn't involve polling. Untested, but how about:

```js
const addRequestCalledTwice = new Promise((resolve) => {
  let callCount = 0;
  const addRequestStub = sinon.stub(Queue.prototype, 'addRequest').callsFake((args) => {
    addRequestStub.wrappedMethod(args);
    if (++callCount >= 2) {
      resolve();
    }
  });
});

// Then, later on:
await addRequestCalledTwice;
```

You'd have to have some way of creating new promises/stubs if you need to wait on it a few different times.